File: src/main/typescript/wcardinal/exception/null-argument-exception.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { Exception } from "./exception";

/**
 * Thrown to indicate that a method has been invoked with a null argument which is not supported.
 *
 *     throw NullArgumentException.create();
 *
 * Must use `isInstance` static method instead of the `instanceof` to check
 * whether an error is an instance of this class.
 * This limitation comes from the fact that the custom exceptions inheriting
 * from the Error class are not printed well in the console.
 *
 *     const e = NullArgumentException.create();
 *
 *     console.log( e instanceof NullArgumentException ); // => false
 *     console.log( NullArgumentException.isInstance( e ) ); // => true
 */
export class NullArgumentException extends Exception {
	constructor() {
		super();
	}

	static getId(): string {
		return "NullArgumentException";
	}

	static getMessage(): string {
		return "unexpected null argument";
	}
}

File: src/main/java/org/wcardinal/exception/NullArgumentException.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.exception;

/**
 * Thrown to indicate that a method has been passed a null argument which is not supported.
 */
public class NullArgumentException extends IllegalArgumentException {
	private static final long serialVersionUID = 4454618571379864803L;

	public NullArgumentException(){
		super();
	}

	public NullArgumentException( final String message ){
		super( message );
	}
}
